lib/archethic/utxo.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Archethic.UTXO do
+  @moduledoc false
+  alias Archethic.DB
+
+  alias Archethic.Crypto
+
+  alias Archethic.Election
+
+  alias Archethic.P2P
+
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.TransactionChain.Transaction.ValidationStamp
+  alias Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperations
+
+  alias Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperations.TransactionMovement
+
+  alias Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperations.UnspentOutput
+
+  alias Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperations.VersionedUnspentOutput
+
+  alias Archethic.Utils
+
+  alias Archethic.UTXO.DBLedger
+  alias Archethic.UTXO.Loader
+  alias Archethic.UTXO.MemoryLedger
+
+  require Logger
+
+  @spec load_transaction(Transaction.t()) :: :ok
+  def load_transaction(tx = %Transaction{}) do
+    authorized_nodes = P2P.authorized_and_available_nodes()
+
+    # Ingest all the movements to fill up the UTXO list
+    ingest_utxo(tx, authorized_nodes)
+
+    # Consume the transaction to update the unspent outputs from the consumed inputs
+    consume_utxo(tx, authorized_nodes)
+
+    Logger.info("Loaded into in memory UTXO tables",
+      transaction_address: Base.encode16(tx.address),
+      transaction_type: tx.type
+    )
+  end
+
+  defp ingest_utxo(
+         %Transaction{
+           address: address,
+           type: tx_type,
+           validation_stamp: %ValidationStamp{
+             protocol_version: protocol_version,
+             timestamp: timestamp,
+             ledger_operations: %LedgerOperations{transaction_movements: transaction_movements}
+           }
+         },
+         authorized_nodes
+       ) do
+    transaction_movements
+    |> Enum.map(&TransactionMovement.maybe_convert_reward_to_uco(&1, tx_type))
+    # |> TransactionMovements.aggregate() after PR about aggregation
```

Comment:
This PR is merged on develop, you can include it

Reply:
Done